### DISCUSSION TITLE
Problems with converting Double to Attempt

### DISCUSSION THREAD

**Discussion by TellSol** [the-anylogic-company/AnyLogic-Pypeline#25] [Q&A] [answered]
Hi, 

I followed the basic tutorial using run() and runResults() in this repo. However, when passing the code:

`Attempt xValue = pyCommunicator.runResults(double.class, "x");`
`traceln(xValue);`

I get _Type mismatch: cannot convert from Double to Attempt_

Any ideas on what the issue can be, and how I can solve it?

**Comment by t-wolfeadam** [comment, marked as answer]
Hello! It seems you're mixing the long vs shortcut way of requesting data. Originally, getting data was a two-step process (assign to an `Attempt`, then request the data type). But I streamlined this in later versions by having the option to declare the class as the first argument to get the result directly, as you did above. In other words, the `Attempt` object is no longer needed and you can just do as follows:

```java
double xValue = pyCommunicator.runResults(double.class, "x");
```

**Reply by TellSol** [reply]
Ok, good. That clears it up. Thanks for the good explanations!
